ibc/relayer.go

```diff
@@ -95,6 +98,78 @@ type Relayer interface {
 	Exec(ctx context.Context, rep RelayerExecReporter, cmd []string, env []string) RelayerExecResult
 }
 
+// GetTransferChannel will return the transfer channel assuming only one client,
+// one connection, and one channel with "transfer" port exists between two chains.
+func GetTransferChannel(ctx context.Context, r Relayer, rep RelayerExecReporter, srcChainID, dstChainID string) (*ChannelOutput, error) {
+	srcClients, err := r.GetClients(ctx, rep, srcChainID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get clients on source chain: %w", err)
+	}
+
+	if len(srcClients) == 0 {
+		return nil, fmt.Errorf("no clients exist on source chain: %w", err)
+	}
+
+	var srcClientID string
+	for _, client := range srcClients {
+		// TODO continue for expired clients
+		if client.ClientState.ChainID == dstChainID {
+			if srcClientID != "" {
+				return nil, fmt.Errorf("found multiple clients on %s tracking %s", srcChainID, dstChainID)
+			}
+			srcClientID = client.ClientID
+		}
+	}
+
+	if srcClientID == "" {
+		return nil, fmt.Errorf("unable to find client on %s tracking %s", srcChainID, dstChainID)
+	}
+
+	srcConnections, err := r.GetConnections(ctx, rep, srcChainID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get connections on source chain: %w", err)
+	}
+
+	if len(srcConnections) == 0 {
+		return nil, fmt.Errorf("no connections exist on source chain: %w", err)
+	}
+
+	var srcConnectionID string
+	for _, connection := range srcConnections {
+		if connection.ClientID == srcClientID {
+			if srcConnectionID != "" {
+				return nil, fmt.Errorf("found multiple connections on %s for client %s", srcChainID, srcClientID)
+			}
+			srcConnectionID = connection.ID
+		}
+	}
+
```

Comment:
Shouldn't we also check if `srcConnectionID == nil` and return an error like we do for the client ID on L124?

Reply:
yes, good catch